src/content/en/updates/2018/07/css-scroll-snap.md

```diff
@@ -0,0 +1,365 @@
+  <section> Section Three </section>
+</article>
+```
+
+
+#### Scroll padding and margin
+
+Our product page has a fixed position top header. Our design also asked for some
+of the top section to remain visible when scroll container is snapped in order
+to provide a design cue to users about the content above.
+
+
+`scroll-padding` is a new css property that can be used to adjust the effective
+viewable region of scroll container.  This region is also known as snapport and
+is used when calculating scroll snap alignments. The property defines an inset
+against the scroll container's padding box. In our example 15vh additional inset
+was added to the top which instructs the browser to consider a lower position,
+15vh below the top edge of the scroll container, as its vertical start edge for
+scroll snapping. When snapping, the start edge of the snap target element will
+become flushed with this new position thus leaving space above.
+
+`scroll-margin` defines the outset amount used to adjust the snap target
+effective box similar to how `scroll-padding` functions on snap scroll
+container.
+
+You may have noticed that these two properties do not have the word "`snap`" in
+them. This is intentional as they actually modify the box for all relevant
+scroll operations and are not just scroll snapping. For example Chrome takes
+them into account when calculating page size for paging scroll operations such
+as PageDown and PageUp and also when calculating scroll amount for
+`Element.scrollIntoView()` operation.
+
+
+<figure>
+  <a href="/web/updates/images/2018/07/css-scroll-snap/product-page.mp4"
+     target="_blank">
+    <video src="/web/updates/images/2018/07/css-scroll-snap/product-page.mp4"
+           autoplay muted playsinline loop controls alt="Product page demo" height="600"></video>
+  </a>
+  <figcaption>
+    <a href="https://snap.glitch.me/product.html"
+       target="_blank">View demo</a> |
+    <a href="https://glitch.com/edit/#!/snap?path=product.html:1:0"
+       target="_blank">Source</a>
+  </figcaption>
+</figure>
+
+## Interaction with other scrolling APIs
+
+### DOM Scrolling API {#dom-scrolling-api}
+
+Scroll snapping happens **after** all scroll operations including those
+initiated by script. When you are using APIs like `Element.scrollTo`, the
+browser will calculate the intended scroll position of the operation, then apply
+appropriate snapping logic to find the final snapped location. Thus,  there is
+no need for user script to do any manual calculations for snapping.
+
+
+### Smooth Scrolling  {#smooth-scrolling}
+
+Smooth scrolling controls the behavior of a programmatic scroll operation while
+scroll snap determines its destination. Since they control orthogonal aspects of
+scrolling, they can be used together and complement each other.
+
+
+### Overscroll Behavior {#overscroll-behavior}
+
+[Overscroll behavior
+API](https://developers.google.com/web/updates/2017/11/overscroll-behavior)
```

Comment:
Change the URL to just the absolute path, eg: `/web/updates/2017/11/overscroll-behavior`

DevSite has other domains (eg developers.google.cn), so to make sure our content is always available, we don't include to host.

Reply:
Done